src/test/java/io/confluent/connect/jdbc/source/integration/JdbcSourceConnectorIT.java

```diff
@@ -0,0 +1,339 @@
+    Client client = new Client();
+    List<ConnectorStateInfo.TaskState> list = connect.connectorStatus(CONNECTOR_NAME).tasks();
+    int id = list.get(0).id();
+    url = url + String.format("/%s/restart", String.valueOf(id));
+    HttpPost postRequest = new HttpPost(url);
+    postRequest.addHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+    HttpResponse httpResponse = client.getHttpClient().execute(postRequest);
+    ConsumerRecords<byte[], byte[]> newRecords = connect.kafka().consume(
+            NUM_RECORDS,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+    Assert.assertEquals(NUM_RECORDS, newRecords.count());
+
+    sendTestDataToMysql(start);
+
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    Thread.sleep(1000);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> newfRecords = connect.kafka().consume(
+            NUM_RECORDS*2,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+    Assert.assertEquals(NUM_RECORDS*2, newfRecords.count());
+
+//    client.getHttpClient().execute()
+  }
+
+  @Test
+  public void testOffsetPickup() throws Exception {
+    int start = 0;
+    String url = connect.endpointForResource(String.format("connectors/%s/tasks", CONNECTOR_NAME));
+    connect.kafka().createTopic("sql-" + KAFKA_TOPIC);
+    dropTableIfExists(KAFKA_TOPIC);
+    createTable();
+    sendTestDataToMysql(start);
+    start += NUM_RECORDS;
+    // Add mysql dialect related configurations.
+    props.put("connection.password", "password");
+    props.put("connection.user", "user");
+    props.put("connection.url", "jdbc:mysql://localhost:3307/db");
+    props.put("dialect.name", "MySqlDatabaseDialect");
+    props.put("value.converter", JsonConverter.class.getName());
+    props.put("mode", "incrementing");
+    props.put("incrementing.column.name",  "id");
+    props.put("topic.prefix",  "sql-");
+
+    // start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expectedNumTasks = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expectedNumTasks);
+
+    Thread.sleep(3000);
+
+    log.info("Waiting for records in destination topic ...");
+    ConsumerRecords<byte[], byte[]> records = connect.kafka().consume(
+            NUM_RECORDS,
+            CONSUME_MAX_DURATION_MS,
+            "sql-mysqlTable");
+    Assert.assertEquals(NUM_RECORDS, records.count());
+    connect.deleteConnector(CONNECTOR_NAME);
+
+// start a source connector
+    connect.configureConnector(CONNECTOR_NAME, props);
+
+    // wait for tasks to spin up
+    int expected = Integer.valueOf(MAX_TASKS); // or set to actual number
+    waitForConnectorToStart(CONNECTOR_NAME, expected);
+
+    Thread.sleep(3000);
+
+    ConsumerRecords<byte[], byte[]> newRecords = connect.kafka().consume(
```

Comment:
Assert the records with actual offsets.
Consumption before restart offsets : 0-100
Consumption after restart offsets : 101-200